src/client/websocket/WebSocketShard.js

```diff
@@ -474,17 +474,23 @@ class WebSocketShard extends EventEmitter {
       this.emit(ShardEvents.ALL_READY);
       return;
     }
+    let time = 15000;
+    if (!new Intents(this.manager.client.options.intents).has(Intents.FLAGS.GUILDS)) {
+      time = 0;
+    }
```

Comment:
```suggestion
    const time = new Intents(this.manager.client.options.intents).has(Intents.FLAGS.GUILDS) ? 15000 : 0;
```
or perhaps use a boolean variable like `hasGuildsIntent`, then below
```js
}, hasGuildsIntent ? 15000 : 0).unref();
```

Reply:
lol, I meant to do it this way but then totally botched it...thanks!